client/block.c

```diff
@@ -526,7 +527,11 @@ static int add_block_nolock(struct xdag_block *newBlock, xdag_time_t limit)
 				}
 				break;
 
-			case XDAG_FIELD_RESERVE0:
+			case XDAG_FIELD_HEAD_TEST:
```

Comment:
XDAG_FIELD_HEAD_TEST can be only in the first field. But first field is skipped in the loop `for(i = 1; i < XDAG_BLOCK_FIELDS; ++i)`
So this case is reduntand.